testsupport/signup_request.go

```diff
@@ -353,3 +365,16 @@ func Close(t *testing.T, resp *http.Response) {
 	err = resp.Body.Close()
 	require.NoError(t, err)
 }
+
+// CreateUserSignupWithSpaceTier creates a UserSignup with space tier specified
+func CreateUserSignupWithSpaceTier(t *testing.T, awaitilities wait.Awaitilities, tier string) (*toolchainv1alpha1.UserSignup, *toolchainv1alpha1.MasterUserRecord) {
+	userSignup, mur := NewSignupRequest(awaitilities).
+		ManuallyApprove().
+		RequireConditions(wait.ConditionSet(wait.Default(), wait.ApprovedByAdmin())...).
+		SpaceTier(tier).
+		EnsureMUR().
+		Execute(t).
+		Resources()
+
+	return userSignup, mur
```

Comment:
let's also wait for the space and return it from the function so there is some added value in the function and it doesn't only abstract the fluent-api.
```go
	space, err := awaitilities.Host().WaitForSpace(t, mur.Name)
	require.NoError(t, err)
	return userSignup, mur, space
```

Reply:
or you could update the SignupRequest API so the `Resources()` function returns also the Space object (if provisioned)

Reply:
Updating returned values of the `Resources()` function of the SignupRequest API would cause many changes across the whole e2e project, so if you decide to do that, then let's do it in a separate PR.

Reply:
I am more inclined to the first option. WDYT?

Reply:
I think I will go with option 2, after all 😄 

Regarding CreateUserSignupWithSpaceTier only abstract the fluent-api, I create this function to avoid duplicated code, since we will need it in other tests like PR #984, PR #985, and PR #990

Reply:
yeah, option 2 is a bit more work, but it will help us with cleaning up some of the duplicated code (retrieval of the Spaces). So I'm for it :+1: 

> Regarding CreateUserSignupWithSpaceTier only abstract the fluent-api, I create this function to avoid duplicated code

I understand that there can be some duplication, but this is something (with the number of test-cases) we just cannot avoid. Also, I have a feeling that the number of helper functions is growing and it's sometimes hard to follow which functions does what and why, so if it was there only to abstract the fluent API, then I would rather drop it and go with the small duplication of the same. In addition to that, it's sometimes better to call the fluent API with the exact values so it's clear what is set and used - like the target member cluster the user is gonna be provisioned to.